client/layout/color-scheme/index.jsx

```diff
@@ -16,6 +16,12 @@ export function getColorScheme( { state, isGlobalSidebarVisible, sectionName } )
 	return siteColorScheme ?? calypsoColorScheme;
 }
 
+export function getColorSchemeFromCurrentQuery( currentQuery ) {
+	return currentQuery?.color_scheme
+		? currentQuery?.color_scheme
+		: new URLSearchParams( currentQuery?.redirect_to ).get( 'color_scheme' );
```

Comment:
If redirect_to is null, undefined, or an invalid URL, URLSearchParams will throw an error. Should we add a fallback for cases where redirect_to is not a valid URL?